commons/Common.ml

```diff
@@ -922,7 +922,7 @@ let cat file =
   cat_aux [] () |> List.rev |> (fun x -> close_in chan; x)
 
 let read_file2 file =
-  let ic = open_in file  in
+  let ic = open_in_bin file  in
   let size = in_channel_length ic in
   let buf = Bytes.create size in
   really_input ic buf 0 size;
```

Comment:
I suspect this could attempt to read beyond the end of the file, presumably resulting in an `End_of_file` exception on Windows.

This is because `in_channel_length` does this:
> The returned size does not take into account the end-of-line translations that can be performed when reading from a channel opened in text mode.

and I assume `really_input` does perform the CRLF->LF conversions because the documentation doesn't say otherwise.

Frankly I don't know which conversions happen and when because (1) the documentation isn't clear about it and (2) I don't have a Windows machine. That's why it's easier to open all files in binary mode.

Source: https://ocaml.org/api/Stdlib.html

Reply:
maybe you should add this comment in the code, so we remember, because I also don't understand anything
about those binary vs textual issue in Windows.

Reply:
What Windows does isn't complicated, it's the OCaml implementation on top of the Windows syscalls that's not obvious.

I'll explain this in `Common.ml` or maybe in a separate readme. What the Windows syscalls do on files opened in text mode is:
* reads: any CRLF (`\r\n`) sequence is translated into LF (`\n`).
* writes: any LF is translated into CRLF.
From that follows a number of rules on how to write our application, I'll detail those as well.

Reply:
I'll also explain the situation with stdin, stdout, and stderr (the open files, not the file descriptors), which are always open in text mode.

Reply:
I changed the behavior of `cat` and added a big explanation in the middle of `Common.ml` because I couldn't think of a better place.